contracts/Token.sol

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.5;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract Token is ERC20, Ownable {
+  address public minter;
+
+  event MinterChanged(address indexed from, address to);
+
+  constructor() ERC20("My Token","MTK") {
+    minter = msg.sender;
+  }
+
+  function passMinterRole(address stakingContract) public onlyOwner() {
+    minter = stakingContract;
+        
+    emit MinterChanged(msg.sender, stakingContract);
+  }
+
+  function mint(address account, uint256 amount) public onlyOwner() {
```

Comment:
you've set a minter, but you're only allowing the owner to mint. You need to use a `minterOnly` modifier. There should be examples of that from open-zeppelin's `AccessControl`